pkg/compact/compact.go

```diff
@@ -547,6 +567,27 @@ func (cg *Group) compact(ctx context.Context, comp tsdb.Compactor) (id ulid.ULID
 	if len(plan) == 0 {
 		return id, nil
 	}
+	// Due to #183 we verify that none of the blocks in the plan have overlapping sources.
+	// This is one potential source of how we could end up with duplicated chunks.
+	uniqueSources := map[ulid.ULID]struct{}{}
+
+	for _, pdir := range plan {
+		meta, err := block.ReadMetaFile(pdir)
+		if err != nil {
+			return id, errors.Wrapf(err, "read meta from %s", pdir)
+		}
+		for _, s := range meta.Compaction.Sources {
+			if _, ok := uniqueSources[s]; ok {
+				return id, halt(errors.Errorf("overlapping sources detected for plan %v", plan))
+			}
+			uniqueSources[s] = struct{}{}
+		}
+		// Ensure all input blocks are valid.
+		if err := block.VerifyIndex(filepath.Join(pdir, "index")); err != nil {
+			return id, errors.Wrapf(halt(err), "invalid plan block %s", pdir)
```

Comment:
This section seems to claim this error: https://github.com/improbable-eng/thanos/issues/246
Let's revisit this code to investigate.  Looks like index is not broken, just index file is no longer there.
@fabxc 

Will have time for it tomorrow.